tests/drink/src/psp22_tests.rs

```diff
@@ -0,0 +1,77 @@
+use anyhow::Result;
+use assert2::assert;
+use drink::{runtime::MinimalRuntime, session::Session};
+use ink_primitives::AccountId;
+use ink_wrapper_types::{util::ToAccountId, Connection};
+use psp22_contract::{Instance, PSP22 as _};
+
+use crate::*;
+
+pub fn balance_of(
+    session: &mut Session<MinimalRuntime>,
+    psp22: Instance,
+    account: AccountId,
+) -> u128 {
+    session
+        .query(psp22.balance_of(account))
+        .unwrap()
+        .result
+        .unwrap()
+}
+
+pub fn setup(session: &mut Session<MinimalRuntime>, caller: drink::AccountId32) -> Instance {
+    let _code_hash = session.upload_code(psp22_contract::upload()).unwrap();
+
+    let _ = session.set_actor(caller);
+
+    session
+        .instantiate(Instance::new(1000))
+        .unwrap()
+        .result
+        .to_account_id()
+        .into()
+}
+
+#[test]
+fn test_transfers() -> Result<()> {
+    let mut session: Session<MinimalRuntime> = Session::new().expect("Init new Session");
+    let instance = setup(&mut session, BOB);
+
+    let transfer_amount = 100;
+
+    let alice_balance = balance_of(&mut session, instance, alice().into());
+    let bob_balance = balance_of(&mut session, instance, bob().into());
+
+    let _res = session
+        .execute(instance.transfer(alice().into(), transfer_amount, vec![]))
+        .unwrap();
+
+    assert!(balance_of(&mut session, instance, bob().into()) == bob_balance - transfer_amount);
+    assert!(balance_of(&mut session, instance, alice().into()) == alice_balance + transfer_amount);
```

Comment:
it's weird that linter doesn't shout for `assert_eq`

Reply:
Actually I've had bad exp with this new `assert2` thingy. When comparing `Result<T, E>` and I had `Ok()` on one side and `Err` on another, it wouldn't tell me what are the actual values. Just that the check fails. So I ended up using `a == b` b/c then it does print out more info about the sides.

Reply:
fair point, resolving